### PR TITLE
40K 12 cards + 1 token

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/imotekh_the_stormlord.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Name:Imotekh the Stormlord
+ManaCost:2 B B
+Types:Legendary Artifact Creature Necron
+PT:3/3
+T:Mode$ ChangesZoneAll | Origin$ Graveyard | Destination$ Any | ValidCards$ Artifact.YouOwn | Execute$ TrigToken | TriggerZones$ Battlefield | TriggerDescription$ Phaeron — Whenever one or more artifact cards leave your graveyard, create two 2/2 black Necron Warrior artifact creature token.
+SVar:TrigToken:DB$ Token | TokenAmount$ 2 | TokenScript$ b_2_2_a_necron_warrior | TokenOwner$ You
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | Execute$ TrigPump | TriggerZones$ Battlefield | TriggerDescription$ Grand Strategist — At the beginning of combat on your turn, another target artifact creature you control gets +2/+2 and gains menace until end of turn.
+SVar:TrigPump:DB$ Pump | ValidTgts$ Creature.Other+Artifact+YouCtrl | NumDef$ +2 | NumAtt$ +2 |  TgtPrompt$ Select another target creature you control | KW$ Menace
+DeckHints:Ability$Graveyard
+DeckHas:Ability$Token & Type$Necron|Warrior & Keyword$Menace
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/imotekh_the_stormlord.txt:10]: Necron not needed here

File: forge-gui/res/cardsfolder/upcoming/imotekh_the_stormlord.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Name:Imotekh the Stormlord
+ManaCost:2 B B
+Types:Legendary Artifact Creature Necron
+PT:3/3
+T:Mode$ ChangesZoneAll | Origin$ Graveyard | Destination$ Any | ValidCards$ Artifact.YouOwn | Execute$ TrigToken | TriggerZones$ Battlefield | TriggerDescription$ Phaeron — Whenever one or more artifact cards leave your graveyard, create two 2/2 black Necron Warrior artifact creature token.
+SVar:TrigToken:DB$ Token | TokenAmount$ 2 | TokenScript$ b_2_2_a_necron_warrior | TokenOwner$ You
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | Execute$ TrigPump | TriggerZones$ Battlefield | TriggerDescription$ Grand Strategist — At the beginning of combat on your turn, another target artifact creature you control gets +2/+2 and gains menace until end of turn.
+SVar:TrigPump:DB$ Pump | ValidTgts$ Creature.Other+Artifact+YouCtrl | NumDef$ +2 | NumAtt$ +2 |  TgtPrompt$ Select another target creature you control | KW$ Menace
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/imotekh_the_stormlord.txt:8]: spacing before prompt

File: forge-gui/res/cardsfolder/upcoming/imotekh_the_stormlord.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Name:Imotekh the Stormlord
+ManaCost:2 B B
+Types:Legendary Artifact Creature Necron
+PT:3/3
+T:Mode$ ChangesZoneAll | Origin$ Graveyard | Destination$ Any | ValidCards$ Artifact.YouOwn | Execute$ TrigToken | TriggerZones$ Battlefield | TriggerDescription$ Phaeron — Whenever one or more artifact cards leave your graveyard, create two 2/2 black Necron Warrior artifact creature token.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/imotekh_the_stormlord.txt:5]: at end: `tokens.`

File: forge-gui/res/cardsfolder/upcoming/imotekh_the_stormlord.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+SVar:TrigToken:DB$ Token | TokenAmount$ 2 | TokenScript$ b_2_2_a_necron_warrior | TokenOwner$ You
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | Execute$ TrigPump | TriggerZones$ Battlefield | TriggerDescription$ Grand Strategist — At the beginning of combat on your turn, another target artifact creature you control gets +2/+2 and gains menace until end of turn.
+SVar:TrigPump:DB$ Pump | ValidTgts$ Creature.Other+Artifact+YouCtrl | NumDef$ +2 | NumAtt$ +2 |  TgtPrompt$ Select another target creature you control | KW$ Menace
+DeckHints:Ability$Graveyard
+DeckHas:Ability$Token & Type$Necron|Warrior & Keyword$Menace
+Oracle:Phaeron — Whenever one or more artifact cards leave your graveyard, create two 2/2 black Necron Warrior artifact creature token.\nGrand Strategist — At the beginning of combat on your turn, another target artifact creature you control gets +2/+2 and gains menace until end of turn.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/imotekh_the_stormlord.txt:11]: at end: `tokens.`

File: forge-gui/res/cardsfolder/upcoming/knight_paladin.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Name:Knight Paladin
+ManaCost:5
+Types:Artifact Vehicle
+PT:6/6
+K:Trample
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigDmg1 | TriggerDescription$ Rapid-fire Battle Cannon — When CARDNAME enters the battlefield, it deals 4 damage to each opponent.
+SVar:TrigDmg1:DB$ DealDamage | NumDmg$ 4 | Defined$ Player.Opponent
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/knight_paladin.txt:7]: Defined$ Opponent

File: forge-gui/res/cardsfolder/upcoming/knight_rampager.txt [hunk 6]
```diff
@@ -0,0 +1,12 @@
+Name:Knight Rampager
+ManaCost:4 R
+Types:Artifact Creature Knight
+PT:6/5
+K:Trample
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | Execute$ TrigChoose | TriggerZones$ Battlefield | TriggerDescription$ Frenzied Rampage — At the beginning of combat on your turn, choose an opponent at random. CARDNAME attacks that player this combat if able.
+SVar:TrigChoose:DB$ ChoosePlayer | Defined$ You | Choices$ Player.Opponent | Random$ True | SubAbility$ DBPump
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/knight_rampager.txt:7]: Choices$ Opponent

File: forge-gui/res/cardsfolder/upcoming/knight_rampager.txt [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+SVar:DBPump:DB$ Pump | Defined$ Self | KW$ HIDDEN CARDNAME attacks specific player each combat if able:ChosenPlayer | Duration$ UntilEndOfCombat
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Card.Self | Execute$ TrigChooseBis | TriggerDescription$ When CARDNAME dies, it deals 4 damage to target opponent chosen at random.
+SVar:TrigChooseBis:DB$ ChoosePlayer | Defined$ You | Choices$ Player.Opponent | Random$ True | SubAbility$ DBDealDamage
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/knight_rampager.txt:10]: Choices$ Opponent

File: forge-gui/res/cardsfolder/upcoming/knight_rampager.txt [hunk 8]
```diff
@@ -0,0 +1,12 @@
+Name:Knight Rampager
+ManaCost:4 R
+Types:Artifact Creature Knight
+PT:6/5
+K:Trample
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | Execute$ TrigChoose | TriggerZones$ Battlefield | TriggerDescription$ Frenzied Rampage — At the beginning of combat on your turn, choose an opponent at random. CARDNAME attacks that player this combat if able.
+SVar:TrigChoose:DB$ ChoosePlayer | Defined$ You | Choices$ Player.Opponent | Random$ True | SubAbility$ DBPump
+SVar:DBPump:DB$ Pump | Defined$ Self | KW$ HIDDEN CARDNAME attacks specific player each combat if able:ChosenPlayer | Duration$ UntilEndOfCombat
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/knight_rampager.txt:8]: add a sub Cleanup to tidy up chosen player

File: forge-gui/res/cardsfolder/upcoming/knight_rampager.txt [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Card.Self | Execute$ TrigChooseBis | TriggerDescription$ When CARDNAME dies, it deals 4 damage to target opponent chosen at random.
+SVar:TrigChooseBis:DB$ ChoosePlayer | Defined$ You | Choices$ Player.Opponent | Random$ True | SubAbility$ DBDealDamage
+SVar:DBDealDamage:DB$ DealDamage | Defined$ ChosenPlayer | NumDmg$ 4
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/knight_rampager.txt:11]: add sub to same cleanup

File: forge-gui/res/cardsfolder/upcoming/knight_rampager.txt [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+Types:Artifact Creature Knight
+PT:6/5
+K:Trample
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | Execute$ TrigChoose | TriggerZones$ Battlefield | TriggerDescription$ Frenzied Rampage — At the beginning of combat on your turn, choose an opponent at random. CARDNAME attacks that player this combat if able.
+SVar:TrigChoose:DB$ ChoosePlayer | Defined$ You | Choices$ Opponent | Random$ True | SubAbility$ DBPump
+SVar:DBPump:DB$ Pump | Defined$ Self | KW$ HIDDEN CARDNAME attacks specific player each combat if able:ChosenPlayer | Duration$ UntilEndOfCombat | SubAbility$ DBCleanup
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Card.Self | Execute$ TrigChooseBis | TriggerDescription$ When CARDNAME dies, it deals 4 damage to target opponent chosen at random.
+SVar:TrigChooseBis:DB$ ChoosePlayer | Defined$ You | Choices$ Opponent | Random$ True | SubAbility$ DBDealDamage
+SVar:DBDealDamage:DB$ DealDamage | Defined$ ChosenPlayer | NumDmg$ 4 | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearChosen$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/knight_rampager.txt:12]: param is named ClearChosenPlayer$

File: forge-gui/res/cardsfolder/upcoming/plasmancer.txt [hunk 11]
```diff
@@ -0,0 +1,8 @@
+Name:Plasmancer
+ManaCost:2 B B
+Types:Artifact Creature Necron Wizard
+PT:3/3
+K:Flying
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigChange | TriggerDescription$ Dynastic Advisor — When CARDNAME enters the battlefield, search your library for a basic Swamp card, reveal it, put it into your hand, then shuffle.
+SVar:TrigChange:DB$ ChangeZone | Origin$ Library | Destination$ Hand | ChangeType$ Swamp.Basic
+Oracle:Flying\Dynastic Advisor — When Plasmancer enters the battlefield, search your library for a basic Swamp card, reveal it, put it into your hand, then shuffle.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/plasmancer.txt:8]: \n

File: forge-gui/res/cardsfolder/upcoming/imotekh_the_stormlord.txt [hunk 12]
```diff
@@ -0,0 +1,11 @@
+Name:Imotekh the Stormlord
+ManaCost:2 B B
+Types:Legendary Artifact Creature Necron
+PT:3/3
+T:Mode$ ChangesZoneAll | Origin$ Graveyard | Destination$ Any | ValidCards$ Artifact.YouOwn | Execute$ TrigToken | TriggerZones$ Battlefield | TriggerDescription$ Phaeron — Whenever one or more artifact cards leave your graveyard, create two 2/2 black Necron Warrior artifact creature tokens.
+SVar:TrigToken:DB$ Token | TokenAmount$ 2 | TokenScript$ b_2_2_a_necron_warrior | TokenOwner$ You
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | Execute$ TrigPump | TriggerZones$ Battlefield | TriggerDescription$ Grand Strategist — At the beginning of combat on your turn, another target artifact creature you control gets +2/+2 and gains menace until end of turn.
+SVar:TrigPump:DB$ Pump | ValidTgts$ Creature.Other+Artifact+YouCtrl | NumDef$ +2 | NumAtt$ +2 | TgtPrompt$ Select another target artifact creature you control | KW$ Menace
+DeckHints:Ability$Graveyard
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/imotekh_the_stormlord.txt:9]: probably add `& Type$Artifact`

File: forge-gui/res/cardsfolder/upcoming/knight_rampager.txt (hunk 10 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/knight_rampager.txt:12]: Might be fun on a rainy day to refactor to allow for stuff like `ClearChosen$ Player,Color` 😁

File: forge-gui/res/cardsfolder/upcoming/lychguard.txt [hunk 13]
```diff
@@ -0,0 +1,8 @@
+Name:Lychguard
+ManaCost:2 B
+Types:Artifact Creature Necron
+PT:2/3
+A:AB$ ChangeZoneAll | Cost$ 3 B Sac<1/CARDNAME> | ChangeType$ Creature.Legendary+YouOwn | PrecostDesc$ Guardian Patrols | Origin$ Graveyard | Destination$ Hand | SpellDescription$ Return all legendary creature cards from your graveyard to your hand.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/lychguard.txt:5]: `Guardian Patrols` > `Guardian Patrols —`

File: forge-gui/res/cardsfolder/upcoming/lychguard.txt [hunk 14]
```diff
@@ -0,0 +1,8 @@
...
+DeckHas:Ability$Graveyard
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/lychguard.txt:6]: `|Sacrifice`

File: forge-gui/res/cardsfolder/upcoming/lychguard.txt [hunk 15]
```diff
@@ -0,0 +1,8 @@
...
+DeckHints:Type$Legendary
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/lychguard.txt:7]: hmmm... maybe `Needs` ?

File: forge-gui/res/cardsfolder/upcoming/knight_rampager.txt [hunk 16]
```diff
@@ -0,0 +1,13 @@
+Name:Knight Rampager
+ManaCost:4 R
+Types:Artifact Creature Knight
+PT:6/5
+K:Trample
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | Execute$ TrigChoose | TriggerZones$ Battlefield | TriggerDescription$ Frenzied Rampage — At the beginning of combat on your turn, choose an opponent at random. CARDNAME attacks that player this combat if able.
+SVar:TrigChoose:DB$ ChoosePlayer | Defined$ You | Choices$ Opponent | Random$ True | SubAbility$ DBPump
+SVar:DBPump:DB$ Pump | Defined$ Self | KW$ HIDDEN CARDNAME attacks specific player each combat if able:ChosenPlayer | Duration$ UntilEndOfCombat | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/knight_rampager.txt:8]: You can't do cleanup for this one or you'll break the attack requirement?

**Review by tool4ever — APPROVED** [review]

**Pull request merged**